test/metabase/query_processor_test/expressions_test.clj

```diff
@@ -449,6 +449,240 @@
+      (is (= [[1 "" "foo" 0 12345 1.234 true false]
+              [2 "" "foo" 0 12345 1.234 true false]]
+             (mt/formatted-rows
+              standard-literal-expression-row-formats-with-id
+              (mt/run-mbql-query orders
+                {:expressions standard-literal-expression-defs
+                 :fields      (into [$id] standard-literal-expression-refs)
+                 :order-by    [[:asc  $id]]
+                 :limit       2})))))))
+
+(deftest ^:parallel filter-literal-expression-with-=-!=-test
+  (doseq [[and-or eq-ne expected] [[:and :=  [standard-literal-expression-values]]
+                                   [:or  :!= []]]]
+    (testing (str "filter literal expressions with " and-or " " eq-ne)
+      (mt/test-drivers (mt/normal-drivers-with-feature :expressions :expression-literals)
+        (is (= expected
+               (mt/formatted-rows
+                standard-literal-expression-row-formats
+                (mt/run-mbql-query orders
+                  {:expressions standard-literal-expression-defs
+                   :fields      standard-literal-expression-refs
+                   :filter      (into [and-or] (map #(vector eq-ne % %) standard-literal-expression-refs))
+                   :limit       1}))))))))
+
+(deftest ^:parallel nested-literal-expression-test
+  (testing "nested literal expression"
+    ;; TODO Fix this test for H2 (QUE-726)
+    (mt/test-drivers (mt/normal-drivers-with-feature :expressions :expression-literals :nested-queries)
+      (is (= [(into [1] standard-literal-expression-values)]
+             (mt/formatted-rows
+              standard-literal-expression-row-formats-with-id
+              (mt/run-mbql-query venues
+                {:fields       (into [$id] standard-literal-expression-column-refs)
+                 :source-query {:source-table $$venues
+                                :expressions  standard-literal-expression-defs
+                                :fields       (into [$id] standard-literal-expression-refs)}
+                 :order-by     [[:asc $id]]
+                 :limit        1})))))))
+
+(deftest ^:parallel order-by-integer-literal-expression-test
+  (testing "order-by integer literal expression"
+    ;; Verify that :order-by of [:expression "One"] does NOT mean ORDER BY 1, which would result in ordering by the
+    ;; first column in the select list $id. We want this to mean "order by the expression with value 1".
+    (mt/test-drivers (mt/normal-drivers-with-feature :expressions :expression-literals :nested-queries)
+      (is (= [[29 "20th Century Cafe" 1]
+              [8  "25°"               1]]
+             (mt/rows
+              (mt/run-mbql-query venues
+                {:expressions {"One" [:value 1 nil]}
+                 :fields      [$id $name [:expression "One"]]
+                 :order-by    [[:asc $name]
+                               [:asc [:expression "One"]]]
```

Comment:
Don't these need to go the other way around for a good test? This is sorting by name and breaking ties with the expression, but there's no tie to break here.

Reply:
Yes, good catch! Fixed.